pkg/globalconfig/global_config.go

```diff
@@ -440,6 +455,75 @@ func WriteGlobalConfig(config GlobalConfig) error {
 	return nil
 }
 
+// ReadGlobalProjects reads the global projects file into DdevProjectList
+// Or creates the file
+func ReadGlobalProjects() error {
+	globalProjectsFile := GetProjectListPath()
+
+	// Can't use fileutil.FileExists() here because of import cycle.
+	if _, err := os.Stat(globalProjectsFile); err != nil {
+		// ~/.ddev doesn't exist and running as root (only ddev hostname could do this)
+		// Then create global projects list.
+		if os.Geteuid() == 0 {
+			logrus.Warning("Not reading global projects file because running with root privileges")
+			return nil
+		}
+		if os.IsNotExist(err) {
+			DdevProjectList = make(map[string]*ProjectInfo)
+			err := WriteGlobalProjects(DdevProjectList)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	source, err := os.ReadFile(globalProjectsFile)
+	if err != nil {
+		return fmt.Errorf("unable to read DDEV global projects file %s: %v", source, err)
+	}
+
+	// ReadConfig config values from file.
+	err = yaml.Unmarshal(source, &DdevProjectList)
+	if err != nil {
+		return err
+	}
+
+	// For backwards compatability we're keeping the project_list in global config
+	// in sync with the list in project_list.yaml. If someone upgrades from an earlier
+	// version the global config will have correct content that isn't in projects.yaml.
+	// For now, treat global config as the source of truth when the two lists differ.
+	if !reflect.DeepEqual(DdevGlobalConfig.ProjectList, DdevProjectList) {
+		DdevProjectList = DdevGlobalConfig.ProjectList
+	}
+
+	return nil
+}
+
+// WriteGlobalProjects writes the global projects list into ~/.ddev.
+func WriteGlobalProjects(projects map[string]*ProjectInfo) error {
+	// Write to global config for backwards compatability.
+	// This allows devs to downgrade to an earlier version without
+	// worrying about copying project info into their global config file.
+	DdevGlobalConfig.ProjectList = projects
+	WriteGlobalConfig(DdevGlobalConfig)
```

Comment:
To pacify the static linter check:

```suggestion
	err := WriteGlobalConfig(DdevGlobalConfig)
	if err != nil {
		return err
	}
```